src/parser/roc_video_parser.h

```diff
@@ -80,6 +80,21 @@ typedef struct {
 #define INIT_SEI_PAYLOAD_BUF_SIZE 1024 * 1024  // initial SEI payload buffer size, 1 MB
 #define DECODE_BUF_POOL_EXTENSION 2
 
+#define CHECK_ALLOWED_RANGE(val, min, max) { \
+    if (val < min || val > max) { \
+        ERR ("value not in range: " + TOSTR(val) + "allowed<min,max>: " + TOSTR(min) + " " + TOSTR(max));\
+        goto error; \
+    } \
+}
+
+#define CHECK_ALLOWED_MAX(val, max) { \
+    if (val > max) { \
+        ERR ("value greater than maximum allowed value: " + TOSTR(val) + " max: " + TOSTR(max));\
+        goto error; \
```

Comment:
As discussed, please replace the `goto error;`  with `return PARSER_OUT_OF_RANGE;`